services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/ResponseSplitterBolt.java

```diff
@@ -41,42 +46,53 @@
 
     @Override
     protected void handleInput(Tuple input) {
-        List<InfoData> responses = (List<InfoData>) input.getValueByField("response");
-        String correlationId = input.getStringByField("correlationId");
+        String correlationId = input.getStringByField(Constants.FIELD_ID_CORELLATION_ID);
         LOGGER.debug("Received response correlationId {}", correlationId);
 
-        sendChunkedResponse(responses, input, correlationId);
+        Object response = input.getValueByField(Constants.FIELD_ID_RESPONSE);
+        List<Message> messages = Collections.emptyList();
+        if (response instanceof List) {
```

Comment:
Well, I assume that `ResponseSplitterBolt` will collect 2 types of errors.
- Errors from Speaker, `RouterBolt` will route it directly to `ResponseSplitterBolt`.
- Errors from `*OperationsBolt`, in case of exception in bolt logic.

For following the design I can propose:
```
kilda.topo.nb-spout ---> router-bolt ---> switches-operations-bolt   ---> response-router-bolt ---> response-splitter-bolt ---> nb-kafka-bolt
                                   | ---> links-operations-bolt      ---> |                  | ---> error-bolt             ---> |
                                   | ---> flows-operations-bolt      ---> |
                                   | ---> validation-operations-bolt ---> |
```

Reply:
At this moment I don't see the reason to have response router, but... it can be useful.

Is any "operations" bolt can produce "list" of error responses? If not, why for we should "split" this kind of response? As far as I remember there can be or error response(it can't even be chained at this moment) or "list"(chain) of results. So error response can be send bypassing "splitter".

Reply:
Currently Bolts can produce a list of `InfoData`, not `ErrorData`. And `ResponseSplitterBolt` works only with list of `InfoData`. For now `*OperationsBolt` not produce errors, but it's planned.